variables.tf

```diff
@@ -9,3 +13,257 @@ variable "tags" {
+  description = "Additional IAM policies to attach to the  ASG instance profile"
+  type        = list(string)
+}
+
+variable "asg_additional_security_groups" {
+  default     = []
+  description = "Additional security group IDs to attach to ASG instances"
+  type        = list(string)
+}
+
+variable "asg_additional_user_data" {
+  default     = ""
+  description = "Additional User Data to attach to the launch template"
+  type        = string
+}
+
+variable "asg_allow_outbound_egress" {
+  default     = true
+  description = "whether or not the default SG should allow outbound egress"
+  type        = bool
+}
+
+variable "asg_desired_capacity" {
+  default     = 1
+  description = "The number of Amazon EC2 instances that should be running in the group."
+  type        = number
+}
+
+variable "asg_instance_type" {
+  default     = "t3a.micro"
+  description = "Instance type for scim app"
+  type        = string
+}
+
+variable "asg_key_name" {
+  default     = null
+  description = "Optional keypair to associate with instances"
+  type        = string
+}
+
+variable "asg_max_size" {
+  default     = 2
+  description = "Maximum number of instances in the autoscaling group"
+  type        = number
+}
+
+variable "asg_min_size" {
+  default     = 1
+  description = "Minimum number of instances in the autoscaling group"
+  type        = number
+}
+
+variable "asg_root_volume_size" {
+  default     = 20
+  description = "size of root volume (includes app install but not data dir)"
+  type        = number
+}
+
+variable "asg_subnet" {
+  description = "Subnet to associate ASG instances with (specify no more than 1)"
+  type        = string
+}
+
+#################################################
+# DB Settings
+#################################################
+
+variable "db_additional_security_groups" {
+  default     = []
+  description = "SGs permitted access to RDS"
+  type        = list(string)
+}
+
+variable "db_allowed_access_cidrs" {
+  default     = []
+  description = "CIDRs permitted access to RDS"
+  type        = list(string)
+}
+
+variable "db_engine_version" {
+  default     = "11"
+  description = "engine version to run"
+  type        = string
+}
+
+variable "db_storage_size" {
+  description = "Size in DB (in GB)"
+  type        = number
+}
+
+variable "db_instance_class" {
+  description = "DB Instance Size"
+  type        = string
+}
+
+variable "db_multi_az" {
+  default     = false
+  description = "If true, DB will be configured in multi-AZ mode"
+  type        = bool
+}
+
+variable "db_monitoring_role_arn" {
+  default     = null
+  description = "ARN for Database Monitoring (required for performance insights)"
+  type        = string
+}
+
+variable "db_monitoring_interval" {
+  default     = 0
+  description = "Enhanced monitoring interval (5-60 seconds, 0 to disable)"
+  type        = number
+}
+
+variable "db_parameters" {
+  description = "DB parameters (downstream module defaults will be used if not specified)"
```

Comment:
downstream module defaults being used seems inaccurate since you've actually specified a non-null default